src/classad/source.cpp

```diff
@@ -68,10 +68,15 @@ ParseExpression( const string &buffer, ExprTree *&tree, bool full )
 {
 	bool              success;
 	StringLexerSource lexer_source(&buffer);
+	tree = nullptr;
```

Comment:
Do we want to assert that the incoming `ExprTree`  is already null?  Or delete it if it isn't?

Reply:
Some callers that put an uninitialized ExprTre* on the stack and pass that in, so no.

Reply:
Also, I plan to remove these methods in 24.x.